### PR TITLE
Complete ArchieML ↔ Raw blocks ↔ Enriched blocks transformation and add tests

### PR DESCRIPTION
**Pull request by danyx23**
So far our GDocs pipeline could go from `ArchieML -> Raw blocks -> Enriched blocks` and from `Raw blocks -> ArchieML`. For generating google docs we need to complete the symmetry so that during backporting we can construct enriched blocks and go down to ArchieML - i.e. the `Enriched blocks -> Raw blocks` step was still missing.

This PR adds that and then also adds some "poor mans property based tests" where I created one example value for each enriched block and then test if converting it all the way to ArchieML syntax and back again to enriched blocks gives us the same enriched block as the one we started with. There were a bunch of issues that broke that assumption and this PR fixes all of them so that we now have proper round-tripping for all current components.

### CODE REVIEW

File: db/model/Gdoc/rawToEnriched.ts [hunk 1]
```diff
@@ -482,7 +484,11 @@ const parseNumberedList = (
             message: "Value is a string, not a list of strings",
         })
 
-    const items = raw.value.map(htmlToEnrichedTextBlock)
+    // ArchieML only has lists, not numbered lists. By convention,
```
Comment (mlbrgl) [inline db/model/Gdoc/rawToEnriched.ts:487]: Was there anything you wanted to add here?

File: db/model/Gdoc/rawToEnriched.ts [hunk 2]
```diff
@@ -829,7 +835,12 @@ function parseStickyRight(
         parseErrors: [error],
     })
     const { left, right } = raw.value
-    if (!left.length || !right.length) {
+    if (
+        left === undefined ||
+        right === undefined ||
+        !left.length ||
+        !right.length
+    ) {
```
Comment (mlbrgl) [inline db/model/Gdoc/rawToEnriched.ts:843]: `!left?.length` ? Or is there the chance to get `left === null` and you want to throw in that case?

**Review by mlbrgl — APPROVED** [review]
Really nice to see the full loop completed, and how this helped uncover some asymetries. The round trip test is really helpful also in getting the high level view of the whole process. Very nice!

**Review by mlbrgl — APPROVED** [review]
Makes sense to me!

**Review by ikesau — APPROVED** [review]
Nice! Ran locally and it all works.

I wonder, can/should we type `enrichedBlockExamples` such that it will throw a type error if we add a new Block type without writing a test for it?

```
const = enrichedBlockExamples: Record<OwidEnrichedArticleBlock["type"], OwidEnrichedArticleBlock> = {
  // etc
}

it.each(Object.values(enrichedBlockExamples))(
  "Parse <-> Serialize roundtrip should be equal - example type $type",
  (example) => {
    // etc
  }
)


```

**Pull request merged**